uber/site_sections/groups.py

```diff
@@ -1,4 +1,101 @@
+    attendee badge.
+    """
+    return attendee.badge_type == c.ATTENDEE_BADGE \
+        and _is_attendee_disentangled(attendee)
+        # It looks like a lot of dealers have helpers that didn't get assigned
+        # a dealer ribbon. We still want to convert those badges, so we can't
+        # trust they'll have a dealer ribbon. I think this is safe because we
+        # won't even get this far if it isn't a dealer group in the first place
+        # and c.DEALER_RIBBON in attendee.ribbon_ints
+
+
+def _decline_and_convert_dealer_group(session, group, convert=False):
+    """
+    Deletes the waitlisted dealer group and converts all of the group members
+    to the appropriate badge type. Unassigned, unpaid badges will be deleted.
+    """
+    admin_note = 'Converted badge from waitlisted dealer group "{}".'.format(
+        group.name)
+
+    if not group.is_unpaid:
+        group.tables = 0
+        for attendee in group.attendees:
+            attendee.append_admin_note(admin_note)
+            attendee.ribbon = remove_opt(attendee.ribbon_ints, c.DEALER_RIBBON)
+        return 'Group dealer status removed'
+
+    message = ['Group declined']
+    emails_failed = 0
+    emails_sent = 0
+    badges_converted = 0
+    badges_deleted = 0
+
+    group.leader = None
+    for attendee in list(group.attendees):
+        if (not convert or attendee.is_unassigned) \
+                and _is_attendee_disentangled(attendee):
+            session.delete(attendee)
+            badges_deleted += 1
+
+        else:
+            if _is_dealer_convertable(attendee):
+                attendee.badge_status = c.NEW_STATUS
+                attendee.overridden_price = attendee.new_badge_cost
+
+                try:
+                    send_email(c.REGDESK_EMAIL,
+                               attendee.email,
+                               'Do you still want to come to {EVENT_NAME}?',
+                               render('emails/dealers/badge_converted.html', {
+                                   'attendee': attendee,
+                                   'group': group
+                               }), model=attendee)
+                    emails_sent += 1
+                except:
+                    emails_failed += 1
+
+            badges_converted += 1
+
+            if attendee.paid not in [c.HAS_PAID, c.NEED_NOT_PAY]:
+                attendee.paid = c.NOT_PAID
+
+            attendee.append_admin_note(admin_note)
+            attendee.ribbon = remove_opt(attendee.ribbon_ints, c.DEALER_RIBBON)
+            group.attendees.remove(attendee)
+
+    session.delete(group)
+
+    for count, template in [
+            (badges_converted, '{} badge{} converted'),
+            (emails_sent, '{} email{} sent'),
+            (emails_failed, '{} email{} failed to send'),
+            (badges_deleted, '{} badge{} deleted')]:
+        if count > 0:
+            message.append(template.format(count, pluralize(count)))
```

Comment:
Nicer messages. We end up with something like "Group declined, 4 badges converted, 1 email sent, 1 badge deleted".